app/models/knight.rb

```diff
@@ -8,5 +8,20 @@ def image
     end
   end
 
+  def valid_move?(new_x, new_y)
+    on_board?(new_x, new_y) && (
+      allowed_deltas = [1, 2]
+      coerced_delta_x = allowed_deltas.delete((x_cord - new_x).abs)
+      coerced_delta_y = allowed_deltas.delete((y_cord - new_y).abs)
+      !!(coerced_delta_x && coerced_delta_y)
+    )
```

Comment:
Ah, just seeing this!

If we are having a valid_move? for each of the pieces, is it important to have an abstraction for the valid_move? on each piece? I am having a hard time seeing the bigger picture of when to abstract & not, so any insight you have is very helpful for me! @ilyakrasnov any resources you can throw my way, I would appreciate, too!